tests/package_tests/k8s_test.py

```diff
@@ -0,0 +1,207 @@
+    )
+
+    # Define API key
+    subprocess.check_call(
+        ["kubectl", "create", "secret", "generic", "scalyr-api-key", f"--from-literal=scalyr-api-key={scalyr_api_key}"]
+    )
+
+    # Create configmap
+    subprocess.check_call(
+        [
+            "kubectl", "create", "configmap", "scalyr-config",
+            "--from-literal=SCALYR_K8S_CLUSTER_NAME=ci-agent-k8s-",
+        ]
+    )
+
+    # Modify the manifest for the agent's daemonset.
+    scalyr_agent_manifest_source_path = __SOURCE_ROOT__ / "k8s/scalyr-agent-2.yaml"
+    scalyr_agent_manifest = scalyr_agent_manifest_source_path.read_text()
+
+    # Change the production image name to the local one.
+    scalyr_agent_manifest = re.sub(
+        r"image: scalyr/scalyr-k8s-agent:\d+\.\d+\.\d+",
+        "image: scalyr/scalyr-k8s-agent:k8s_test",
+        scalyr_agent_manifest
+    )
+
+    # Change image pull policy to be able to bull the local image.
+    scalyr_agent_manifest = re.sub(
+        r"imagePullPolicy: \w+", "imagePullPolicy: Never", scalyr_agent_manifest
+    )
+
+    # Create new manifest file for the agent daemonset.
+    tmp_dir = tempfile.TemporaryDirectory(prefix="scalyr-agent-k8s-test")
+
+    scalyr_agent_manifest_path = pl.Path(tmp_dir.name) / "scalyr-agent-2.yaml"
+
+    scalyr_agent_manifest_path.write_text(scalyr_agent_manifest)
+
+    # Create agent's daemonset.
+    subprocess.check_call([
+        "kubectl", "create", "-f", str(scalyr_agent_manifest_path)
+    ])
+
+    # Get name of the created pod.
+    pod_name = subprocess.check_output(
+        "kubectl get pods --sort-by=.metadata.creationTimestamp -o jsonpath=\"{.items[-1].metadata.name}\"", shell=True
+    ).decode().strip()
+
+    # Wait a little.
+    time.sleep(3)
+
+    # Execute tail -f command on the agent.log inside the pod to read its content.
+    agent_log_tail_process = subprocess.Popen(
+        ["kubectl", "exec", pod_name, "--container", "scalyr-agent", "--", "tail", "-f", "-n+1",
+         "/var/log/scalyr-agent-2/agent.log"],
+        stdout=subprocess.PIPE
+    )
+    found_errors = []
+
+    # Read lines from agent.log.
+    pipe_reader = PipeReader(pipe=agent_log_tail_process.stdout)
+
+    timeout_time = datetime.datetime.now() + datetime.timedelta(seconds=COMMON_TIMEOUT)
+    error_lines = []
+    try:
+        while True:
+
+            seconds_until_timeout = timeout_time - datetime.datetime.now()
+            line = pipe_reader.next_line(timeout=seconds_until_timeout.seconds)
+            print(line)
+            # Look for any ERROR message.
+            if check_if_line_an_error(line):
+                error_lines.append(line)
+
+            # TODO: add more checks.
+
+            if check_agent_log_request_stats_in_line(line):
+                # The requests status message is found. Stop the loop.
+                break
+    finally:
+        agent_log_tail_process.terminate()
+
+    agent_log_tail_process.communicate()
```

Comment:
Do we also want to stop ``PipeReader`` thread here?

And I guess it's probably safer to run all the "clean up" routines inside ``finally`` block?

Reply:
I changed the pipe reader, so it doesn't create a thread anymore.